src/main/java/com/microsoft/aad/adal4j/AuthenticationContext.java

```diff
@@ -255,6 +253,54 @@ public AuthenticationResult call() throws Exception {
     }
 
     /**
+     * Acquires a security token from authority using an authenticated client with password credential
+     * grant.
+     *
+     * @param resource
+     *            Identifier of the target resource that is the recipient of the
+     *            requested token. If null, token is requested for the same
+     *            resource refresh token was originally issued for. If passed,
+     *            resource should match the original resource used to acquire
+     *            refresh token unless token service supports refresh token for
+     *            multiple resources.
+      @param credential
+     *            object representing Private Key to use for token acquisition.
+     * @param username
+     *            Username of the managed or federated user.
+     * @param password
+     *            Password of the managed or federated user.
+     * @param callback
+     *            optional callback object for non-blocking execution.
+     * @return A {@link Future} object representing the
+     *         {@link AuthenticationResult} of the call. It contains Access
+     *         Token, Refresh Token and the Access Token's expiration time.
+     */
+    public Future<AuthenticationResult> acquireToken(final String resource,

```

Comment:
in this flow you exchange a refresh token for a resource 1 with an access token for resource 2, but my use case is a bit different:
- A native client acquires an access token for a recuse in a automated manner (this client is used in integration tests)
- The native client passes the token to a resources to test the authentication and authorization checks for a particular user. This means the token needs to contain user claims that the verification is effective. So I can not use the client credential flow. 
- The resource is registered with  login.microsoft.com authority using the AAD on boring process which unlike login.microsoftonline.com does not support public client 

I am forced to use a confidential client, because of the last point, otherwise I wouldn't be a problem to use a public client.

Do you have any suggestions. I would appreciate if you could consider our use case in your roadmap.